Moonshot/player/player.gd

```diff
@@ -157,24 +160,33 @@ func set_invulnerable(time : float, animation_name = "stagger") -> void:
 	safety = false
 	
 
-func damagetile_check() -> void:
+func damagetile_check():
 	for i in self.get_slide_count():
 		var collision = self.get_slide_collision(i)
 		if !collision || !collision.collider:
 			break
 			
 		if collision.collider.name == "DamagePools" or collision.collider.name == "DamageWalls":
-			if is_on_floor():
-				take_damage(0.1,Vector2.UP, true)
-				
-			elif is_on_wall():
-				var wall_normal = $LedgeWallDetector.scale.x * - 1
-				take_damage(0.1,Vector2(wall_normal,0) , true)
+			return true
+		else:
+			return false
+
+func damagetile_impulse():
+	$SFX.play("sizzle")
+	if is_on_floor():
+		take_damage(0.1,Vector2.UP, true)
+		
+	elif is_on_wall():
+		var wall_normal = $LedgeWallDetector.scale.x * - 1
+		take_damage(0.1,Vector2(wall_normal,0) , true)
```

Comment:
Ah is this a fix for the damage tiles as well? 😄

Reply:
This is mostly just splitting the impule logic from the actual damagetile check. Allowed me to use the check somewhere else as well.